client/Containers/PaginationControl.jsx

```diff
@@ -0,0 +1,136 @@
+
+// const mapDispatchToProps = (dispatch) => ({
+//   setPagination: (productPagination) =>
+//     dispatch(setProductPagination(productPagination)),
+//   paginationFuncs: {
+//     nextPage: () => dispatch(nextPage()),
+//     prevPage: () => dispatch(prevPage()),
+//     goToPage: (page) => dispatch(goToPage(page)),
+//     sizePage: (size) => {
+//       dispatch(sizePage(size));
+//       dispatch(goToPage(1));
+//     },
+//     sortPage: (sort) => dispatch(sortPage(sort)),
+//   },
+// });
+
+const PaginationBar = styled.div`
+  display: flex;
+  justify-content: space-between;
+  background: black;
+  width: calc(100%);
+  border-radius: ${(props) => (props.top ? '0.5rem 0.5rem 0 0' : '0.5rem')};
+`;
+
+const PageOptionContainer = styled.div`
+  display: flex;
+`;
+
+const PageOption = styled.div`
+  padding: 1em;
+  color: ${(props) =>
+    props.operable ? 'white' : props.current ? '#ffda08' : 'gray'};
+  cursor: ${(props) => (props.operable ? 'pointer' : 'default')};
+`;
+
+const PaginationControl = ({ top }) => {
+  const { productFilters, productPagination } = useSelector(
+    ({ productFilters, productPagination }) => ({
+      productFilters,
+      productPagination,
+    })
+  );
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const getProducts = (query) => dispatch(fetchProductList(query));
+  const paginate = (options) => {
+    // options: { page?: INT, size?: INT, sort?: STRING }
+    for (const key in productPagination) {
+      if (options[key]) productPagination[key] = options[key];
+    }
+    const query = buildProductQuery({ productFilters, productPagination });
+    history.push(`/products${query}`);
+    getProducts(query);
+  };
+
+  const { maxPage, page, size, sort } = productPagination;
+  const [sortBy, sortDirection] = sort.split(',');
+  return (
+    <PaginationBar top={top}>
+      <PageOptionContainer>
+        <PageOption
+          operable={page > 1}
+          onClick={() => {
+            if (page > 1) paginate({ page: 1 });
+          }}
+        >
+          First
+        </PageOption>
+        <PageOption
+          operable={page > 1}
+          onClick={() => {
+            if (page > 1) paginate({ page: page - 1 });
+          }}
+        >
+          Previous
+        </PageOption>
+      </PageOptionContainer>
+      <PageOptionContainer>
+        {Array(maxPage)
+          .fill('')
+          .map((val, idx) => {
+            const pageOption = idx + 1;
+            const currPageCheck = page !== pageOption;
+            return (
+              <PageOption
+                key={pageOption}
+                operable={currPageCheck}
+                current={!currPageCheck}
+                onClick={() => {
+                  if (currPageCheck) paginate({ page: pageOption });
+                }}
+              >
+                {pageOption}
+              </PageOption>
+            );
+          })}
```

Comment:
Adding a comment to this file now, but we probably want to limit how many specific pages there are available to the user, say 10?